src/utilities/sequence-editor/sequence-linter.ts

```diff
@@ -1185,13 +1177,14 @@ function validateArgument(
           break;
         }
         const { max, min } = dictArg.range;
-        const nodeTextAsNumber = parseFloat(argText);
-
+        const nodeTextAsNumber = parseNumericArg(argText, dictArgType);
+        const isHex = isHexValue(argText);
         if (nodeTextAsNumber < min || nodeTextAsNumber > max) {
+          const numFormat = (n: number) => (isHex ? `0x${n.toString(16).toUpperCase()}` : n);
```

Comment:
Can this function be broken out to a dedicated function? I know it was just moved to a new spot, but it seems unnecessary to have to have the memory reallocated for it for every argument, when it could be memoized or just declared statically ahead of time.